--- ./setup.py
+++ ./setup.py
@@ -123,7 +123,7 @@
     #
     # For an analysis of "install_requires" vs pip's requirements files see:
     # https://packaging.python.org/en/latest/requirements.html
-    install_requires=['PyYAML==5.1'],  # Optional
+    install_requires=['PyYAML>=6.0'],  # Optional
 
     # List additional groups of dependencies here (e.g. development
     # dependencies). Users will be able to install these using the "extras"
